aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/AwsPresigningDecorator.kt

```diff
@@ -128,21 +141,64 @@ class AwsPresigningDecorator internal constructor(
             }
         }.build()
     }
+
+    private fun anyPresignedShapes(ctx: ClientCodegenContext) =
+        TopDownIndex.of(ctx.model).getContainedOperations(ctx.serviceShape)
+            .any { presignableOperations.containsKey(it.id) }
+
+    override fun extraSections(codegenContext: ClientCodegenContext): List<AdHocCustomization> =
+        anyPresignedShapes(codegenContext).thenSingletonListOf {
+            adhocCustomization<CustomizableOperationSection.CustomizableOperationImpl> {
+                rustTemplate(
+                    """
+                    /// Sends the request and returns the response.
+                    ##[allow(unused_mut)]
+                    pub async fn presigned(mut self, presigning_config: #{PresigningConfig}) -> #{Result}<#{PresignedRequest}, crate::error::SdkError<E>> where
+                        E: std::error::Error + #{Send} + #{Sync} + 'static,
+                        B: #{CustomizablePresigned}<E>
+                    {
+                        let mut config_override = self.config_override.unwrap_or_default();
+                        self.interceptors.into_iter().for_each(|interceptor| {
+                            config_override.push_interceptor(interceptor);
+                        });
+                        self.runtime_plugins.into_iter().for_each(|plugin| {
+                            config_override.push_runtime_plugin(plugin);
+                        });
+
+                        self.customizable_send.presign(config_override, presigning_config).await
+                    }
+                    """,
+                    *codegenScope,
+                    "CustomizablePresigned" to CustomizablePresigned,
+                )
+            }
+        }
+
+    private val CustomizablePresigned =
+        RuntimeType.forInlineFun("CustomizablePresigned", InternalTraitsModule) {
+            rustTemplate(
+                """
+                pub trait CustomizablePresigned<E> {
```

Comment:
How come `CustomizableSend` has `Send + Sync` bounds, but this one doesn't?

Reply:
no good reason—they aren't actually needed because we never make a trait object from it. I added them here for consistency.